src/serialization/portable/ClassDefinitionContext.ts

```diff
@@ -48,9 +51,13 @@ export class ClassDefinitionContext {
             this.classDefs[cdKey] = classDefinition;
             return classDefinition;
         }
-        if (current instanceof ClassDefinition && !current.equals(classDefinition)) {
-            throw new RangeError(`Incompatible class definition with same class id: ${classDefinition.getClassId()}`);
+        try {
+            deepEqual(current, classDefinition);
```

Comment:
I assume `deepEqual` checks private fields that are supposed to be implementation details. This is not good practice. A good way would be to have a public `equals` method in `ClassDefinition` and have it do the comparison.